### PR TITLE
Parse new .xcresult scheme to support testing with Xcode 11

### PR DESCRIPTION
**Pull request by BastianKusserow**
This PR adds the ability to parse the new `.xcresult` file format to get the right test output for Xcode 11.
This also solves #419

### CODE REVIEW

File: libxcode/src/main/groovy/org/openbakery/test/TestResultParser.groovy [hunk 1]
```diff
@@ -48,21 +50,92 @@ public class TestResultParser {
...
-			def identifier = testResult.getString("RunDestination.TargetDevice.Identifier")
+		if (isTestSummaryPlistAvailable()) {
+			logger.debug("parsing xcresult scheme version < 3")
+			testSummariesArray.toList().each {
+				def testResult = new XMLPropertyListConfiguration(it)
+				def identifier = testResult.getString("RunDestination.TargetDevice.Identifier")
+
+				Destination destination = findDestinationForIdentifier(destinations, identifier)
+				if (destination != null) {
+					def resultList = processTestSummary(testResult.getList("TestableSummaries"))
```
Comment (fonkadelic) [inline libxcode/src/main/groovy/org/openbakery/test/TestResultParser.groovy]:
```suggestion
					def resultList = processLegacyTestSummary(testResult.getList("TestableSummaries"))
```

File: libxcode/src/main/groovy/org/openbakery/test/TestResultParser.groovy [hunk 2]
```diff
@@ -48,21 +50,92 @@ public class TestResultParser {
...
+
+		return object
+	}
+
+	ArrayList<TestClass> parseNewTestSummaries(Map<String, Object> xcResult, File file) {
+		def runner = new CommandRunner()
+		def json = new JsonSlurper()
+
+		def testsRef = xcResult.actions._values.actionResult.testsRef.id._value[0]
+		if(testsRef == null) { return }
```
Comment (fonkadelic) [inline libxcode/src/main/groovy/org/openbakery/test/TestResultParser.groovy]: Debugging logging would probably be helpful here.

File: build.gradle [hunk 3]
```diff
@@ -18,8 +18,8 @@ allprojects {
 	group = 'org.openbakery'
 	version = versionNumber
 
-	sourceCompatibility = "1.6"
-	targetCompatibility = "1.6"
+	sourceCompatibility = "1.7"
```
Comment (fonkadelic) [inline build.gradle]: @renep Maybe you have some feedback on this. I don't know the exact consequences this change can cause.

Comment (renep) [reply build.gradle]: Java 1.6 ist quite old. I have no problem to go to 1.8 or higher

Comment (BastianKusserow) [reply build.gradle]: I've set the java version to 1.8 and updated the Readme.

**Comment by renep** [comment]
I have reverted it because I want to merge it into the `develop` branch. My CI build a `develop` version where the changes can be tested before it is officially release.
See #422

**Comment by renep** [comment]
You can test build `0.19.2.develop.155` that contains this merge request.

**Pull request merged**